pycode/memilio-epidata/memilio/epidata_test/test_epidata_geoModificationGermany.py

```diff
@@ -268,7 +327,83 @@ def test_get_nuts3_county_id_map(self, mock_print):
         mock_print.assert_called_with(
             'Source data frame contains more counties than official'
             ' county list. This could be OK, please verify yourself.')
-    
+
+    def test_get_intermediateregion_IDs(self):
+        # zfill is false
+        unique_geo_entitites = geoger.get_intermediateregion_ids(False)
+        self.assertEqual(unique_geo_entitites, self.test_list_regions1)
+
+        # zfill is true
+        unique_geo_entitites = geoger.get_intermediateregion_ids(True)
+        self.assertEqual(unique_geo_entitites, self.test_list_regions2)
+
+    def test_get_intermediateregion_names(self):
+
+        region_names = geoger.get_intermediateregion_names()
+        self.assertEqual(region_names, self.test_list_regions3)
+
+    def get_intermediateregion_to_name(self):
+
+        region_to_name = geoger.get_intermediateregion_to_name()
+        self.assertEqual(region_to_name, dict(zip(self.test_list_regions1,
+                                                  self.test_list_regions3)))
+
+    def test_get_intermediateregion_names_and_ids(self):
+
+        # zfill is false
+        regionnamesandids = geoger.get_intermediateregion_names_and_ids(False)
+        testregionnamesandids = []
+        for i in range(0, len(self.test_list_regions1)):
+            combined = [self.test_list_regions3[i], self.test_list_regions1[i]]
+            testregionnamesandids.append(combined)
+        self.assertEqual(regionnamesandids, testregionnamesandids)
+
+        # zfill is true
+        regionnamesandids = geoger.get_intermediateregion_names_and_ids(True)
+        testregionnamesandids = []
+        for i in range(0, len(self.test_list_regions2)):
+            combined = [self.test_list_regions3[i], self.test_list_regions2[i]]
+            testregionnamesandids.append(combined)
+        self.assertEqual(regionnamesandids, testregionnamesandids)
+
+    def test_get_countyid_to_intermediateregionid_map(self):
+
+        countytoregion = geoger.get_countyid_to_intermediateregionid_map(
+            merge_eisenach=True, zfill=False)
+        self.assertEqual(all(self.countytoregion_string), any(countytoregion))
+        self.assertEqual(any('16063: 6'), any(countytoregion))
+        self.assertNotEqual(('16056: 6'), any(countytoregion))
```

Comment:
I do not really understand what is tested in the three lines above.

Reply:
Me neither ;-)

Changed as discussed.